### ISSUE TITLE
New upload destination when testing

### ISSUE THREAD

**Issue by Nayte91** [dustin10/VichUploaderBundle#1278]
|    Q            |   A
|---------------- | ------
| Bundle version  | 1.19.0
| Symfony version | 6.0.3
| PHP version     | 8.1.3

### Support Question

Hello,
I'm working with Symfony & API-Platform, and I'm redacting functional tests (ApiTestCase) with my uploads.

I usually upload my files in the public/media/foo folder, and for testing (to be able to delete folder's content easily) I want to upload in public/media/test/foo. Config is in packages/vich_uploader.yaml.

I can't figure out why my overrided packages/test/vich_uploader.yaml is not changing my behavior when I use bin/phpunit. Neither the customized upload_destination nor namer keys is working when I launch tests (files are written in prod/dev folder with old/dev namer).

Do you know if this is something impossible ? Am I mixing up with a simple issue ? Am I missing a trick ?

My dumps:

Dev environment -->

```
/app # APP_ENV=dev php bin/console debug:config vich_uploader

Current configuration for extension with alias "vich_uploader"
==============================================================

vich_uploader:
    db_driver: orm
    metadata:
        type: attribute
        cache: file
        file_cache:
            dir: /app/var/cache/dev/vich_uploader
        auto_detection: true
        directories: {  }
    mappings:
        event_banner:
            uri_prefix: /media/banner
            upload_destination: /app/public/media/banner
            namer:
                service: Vich\UploaderBundle\Naming\OrignameNamer
                options: {  }
            directory_namer:
                service: null
                options: null
            delete_on_remove: true
            delete_on_update: true
            inject_on_load: false
            db_driver: null
        event_logo:
            uri_prefix: /media/logo
            upload_destination: /app/public/media/logo
            namer:
                service: Vich\UploaderBundle\Naming\OrignameNamer
                options: {  }
            directory_namer:
                service: null
                options: null
            delete_on_remove: true
            delete_on_update: true
            inject_on_load: false
            db_driver: null
    default_filename_attribute_suffix: _name
    storage: file_system
    templating: false
    twig: true
    form: true
```

Test environment -->

```
/app # APP_ENV=test php bin/console debug:config vich_uploader

Current configuration for extension with alias "vich_uploader"
==============================================================

vich_uploader:
    db_driver: orm
    metadata:
        type: attribute
        cache: file
        file_cache:
            dir: /app/var/cache/test/vich_uploader
        auto_detection: true
        directories: {  }
    mappings:
        event_banner:
            uri_prefix: /media/banner
            upload_destination: /app/public/tests/media/banner
            namer:
                service: Vich\UploaderBundle\Naming\UniqidNamer
                options: {  }
            directory_namer:
                service: null
                options: null
            delete_on_remove: true
            delete_on_update: true
            inject_on_load: false
            db_driver: null
        event_logo:
            uri_prefix: /media/logo
            upload_destination: /app/public/tests/media/logo
            namer:
                service: Vich\UploaderBundle\Naming\UniqidNamer
                options: {  }
            directory_namer:
                service: null
                options: null
            delete_on_remove: true
            delete_on_update: true
            inject_on_load: false
            db_driver: null
    default_filename_attribute_suffix: _name
    storage: file_system
    templating: false
    twig: true
    form: true
```

TL;DR: Test env config isn't overriding when php bin/phpunit.

And if it can help, [here's the repository with files](https://github.com/Nayte91/ephemere_test). main files involved:

- tests/API/EventBannerResourceTest.php
- src/Entity/MediaObject/EventBanner.php
- config/packages/vich_uploader.yaml
- config/packages/test/vich_uploader.yaml

Thank you very much in advance, and thank you for your work ❤️
